### PR TITLE
Don't use GitHub API for project syncs.

### LINKED ISSUE #8565: Getting a lot of "<Project> not found in datastore" issues
https://pantheon.corp.google.com/errors/detail/CIWG-ZGdo_fuHg?project=oss-fuzz
CC @oliverchang

### PR DESCRIPTION
**Pull request by oliverchang**
Download the zip and unzip it instead. There should be enough memory for this on the tmpfs environment on Cloud Functions, but bump up the memory to 4GB just in case.

Create a `OssFuzzRepo` which replicates the GitHub client library to avoid changing too many things. 

Fixes #8565

### CODE REVIEW

**Review by Alan32Liu — APPROVED** [review]
LGTM

**Comment by oliverchang** [comment]
/gcbrun trial_build.py abseil-cpp --sanitizer address --fuzzing-engine libfuzze

**Comment by oliverchang** [comment]
/gcbrun trial_build.py abseil-cpp --sanitizer address --fuzzing-engine libfuzzer

File: infra/build/functions/project_sync.py [hunk 1]
```diff
@@ -43,9 +45,54 @@
 ProjectMetadata = namedtuple(
     'ProjectMetadata', 'schedule project_yaml_contents dockerfile_contents')
 
+Content = namedtuple('Content', 'type path name decoded_content')
```
Comment (jonathanmetzman) [inline infra/build/functions/project_sync.py:48]: Didn't know you can use this syntax for named tuple.

**Review by jonathanmetzman — COMMENTED** [review]
LGTM
Good idea to use the zip

**Pull request merged**